### PR TITLE
Implement properties fetch for GitLab

### PR DESCRIPTION
**Pull request by JAORMX**
# Summary

This does a very basic implementation of fetching the base propertiesf
or a repo in gitlab.

Closes: #4330

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [x] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: internal/providers/gitlab/repository_properties.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	outProps, err := properties.NewProperties(map[string]any{
+		properties.RepoPropertyIsPrivate:  proj.Visibility == gitlab.PrivateVisibility,
+		properties.RepoPropertyIsArchived: proj.Archived,
+		properties.RepoPropertyIsFork:     proj.ForkedFromProject != nil,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create properties: %w", err)
+	}
+
+	return outProps.Merge(getByProps), nil
```
Comment (jhrozek) [inline internal/providers/gitlab/repository_properties.go:80]: should we merge getByProps as-is or just cherry-pick the known values?

Comment (JAORMX) [reply internal/providers/gitlab/repository_properties.go:80]: I don't have a strong opinion, as long as `outProps` takes precedence.

Comment (jhrozek) [reply internal/providers/gitlab/repository_properties.go:80]: what maybe is not apparent: merge replaces any properties in outProps with same-named properties in getByProps so the caller could essentially set the outProps...

Comment (JAORMX) [reply internal/providers/gitlab/repository_properties.go:80]: Ah! So I meant to do the opposite then.

File: internal/providers/gitlab/repository_properties.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+func (c *gitlabClient) getPropertiesForRepo(
+	ctx context.Context, getByProps *properties.Properties,
+) (*properties.Properties, error) {
+	uid, err := getByProps.GetProperty(properties.PropertyUpstreamID).AsString()
+	if err != nil {
+		return nil, fmt.Errorf("upstream ID not found or invalid: %w", err)
+	}
+
+	projectURLPath, err := url.JoinPath("projects", url.PathEscape(uid))
```
Comment (jhrozek) [inline internal/providers/gitlab/repository_properties.go:39]: I guess later we'll split this into a helper when we have more getProperties calls

**Review by jhrozek — APPROVED** [review]

File: internal/providers/gitlab/repository_properties.go (hunk 2 above)
Comment (JAORMX) [reply internal/providers/gitlab/repository_properties.go:39]: yep, all of this will get refactored in due time.

**Pull request merged**